stdlib/public/core/StringGuts.swift

```diff
@@ -201,7 +201,7 @@ extension _StringGuts {
     the runtime is depending on this, update Reflection.mm and \
     this if you change it
     """)
-    #elseif _pointerBitWidth(_32)
+    #elseif _pointerBitWidth(_32) || _pointerBitWidth(_16)
     _internalInvariant(MemoryLayout<String>.size == 12, """
```

Comment:
This is also not going to add up on 16-bit, but I guess that's fine and then once we get to the point where we actually want to get String to work on AVR, we'll inevitably hit this assertion.

Reply:
yeah, that was roughly my thinking... I feel like the standard library could benefit from a _static_assert function. But I suspect many people have said this before and the arguments for and against are as old as time. :)

Reply:
Memory layout will be 8 bytes according to my guesses.

Reply:
But I really like the look of Jordan's approach which gets it down to 6 bytes. In fact, if we can get it to 4 bytes, that's kind of a sweet spot for the AVR back end...